crates/block-producer/src/batch_builder/batch.rs

```diff
@@ -144,18 +172,18 @@ impl TransactionBatch {
     pub fn account_initial_states(&self) -> impl Iterator<Item = (AccountId, Digest)> + '_ {
         self.updated_accounts
             .iter()
-            .map(|(_, update)| (update.account_id(), update.init_state_hash()))
+            .map(|(&account, update)| (account, update.init_state))
```

Comment:
nit: I would probably name the variable `account_id` rather than just `account`.